lib/src/room.dart

```diff
@@ -1100,17 +1100,26 @@ class Room {
           txid: messageID,
         );
       } catch (e, s) {
-        if (e is MatrixException || DateTime.now().isAfter(timeoutDate)) {
+        if (e is MatrixException &&
+            e.retryAfterMs != null &&
+            !DateTime.now().isAfter(timeoutDate)) {
+          Logs().w(
+              'Ratelimited while sending message, waiting for ${e.retryAfterMs}ms');
+          await Future.delayed(Duration(milliseconds: e.retryAfterMs!));
```

Comment:
But after waiting this is not resending the message right? Because it will not go into the other else if block where it tries to send it again but just go to the end of the method if I see this correctly

Reply:
the whole method is wrapped in a while true already, the else branch retries as well, afaik? Did I misread this? :D

Reply:
Sorry, yes you are right. Haven't expanded the focus. What happens when the `e.retryAfterMs` is larger than the `timeoutDate`?

Reply:
I guess then it waits for a too long period and then fails. I guess that could be prettier, if we just immediately fail in that case? However that then again quietly doesn't send the event, so maybe we should propagate the exceptions at some point?